### DISCUSSION TITLE
Unpublished fields in publication

### DISCUSSION THREAD

**Discussion by kletersteegp** [OP-TED/eForms-SDK#513] [Q&A] [answered]
Hello,

I have made a publication example with the SDK and send the same XML to the preview environment and created there a publication example.  On the TED preview environment i see that the field value is shown as unpublished (with the text unpublished).

![image](https://github.com/OP-TED/eForms-SDK/assets/111063502/a899bbd6-6deb-4791-b519-8cbfee27daf4)

But with the SDK i see the same example with the field value. 
![image](https://github.com/OP-TED/eForms-SDK/assets/111063502/015aa471-b422-44e4-9bcd-66aad45b7359)

My question is: on the TED preview environment are you doing something extra (outside the SDK) to show the value as unpublished?  Remark that in the screenshot created purely with the SDK the other unpublished fields (as reason) are shown. And i use the same XML for both screenshots. I am just curious what is the explanation for the difference and how we can make a publication example with the text unpublished on our own platform.

Regards, Peter.

**Comment by rouschr** [comment]
What do you mean by "created purely with the SDK", the SDK is only meta-data + documentation, you mean using the eforms notice viewer?
https://github.com/OP-TED/eforms-notice-viewer/pulls
I do not think the eforms notice viewer has special treatment for unpublished fields.
In the SDK of fields.json you can see:
https://github.com/OP-TED/eForms-SDK/blob/1.7.0/fields/fields.json
Something like (just any example):
```
    "maxLength" : 6000,
    "privacy" : {
      "code" : "awa-cri-des",
      "unpublishedFieldId" : "BT-195(BT-540)-Lot",
      "reasonCodeFieldId" : "BT-197(BT-540)-Lot",
      "reasonDescriptionFieldId" : "BT-196(BT-540)-Lot",
      "publicationDateFieldId" : "BT-198(BT-540)-Lot"
    },
```
This tells you which fields are used to unpublish another field.

**Reply by kletersteegp** [reply]
Hello, we have a simular application as the notice viewer to render the eForms XML to html. If we use that than the unpublished field is shown with the value and then with a reason why it should not be published. But if we sent the XML in to the preview environment then the same field is not shown the value. But is shows as 'unpublished'.  The question is where in your code you give the unpublished field the value 'unpublished'. As shown in the picture below. Where does the word with the circle around it 'unpublished' come from? Where is that implemented in your code that the text 'unpublished' is shown. If we know that then we could do it the same way. 

![image](https://github.com/OP-TED/eForms-SDK/assets/111063502/b2fd42ab-8d12-4fff-9052-32d37ddf5020)

**Comment by kletersteegp** [comment, marked as answer]
We analysed it more at our side. And it looks that the problem is at our side. So this issue can be closed.
